src/Mermaid/props.ts

```diff
@@ -0,0 +1,139 @@
+import mermaidAPI from 'mermaid/mermaidAPI';
+
+/**
+ * This type contains all the configuration of mermaid to be used as properties of the plugin.
+ * @public
+ */
+export type MermaidProps = {
+  config?: {
```

Comment:
Nit: as `config.config` reads a bit strange, maybe name this property `mermaidConfig` so its more obvious that this is the config for mermaid itself and not the plugin.